CONTRIBUTING.md

```diff
@@ -0,0 +1,78 @@
+<!---
+   Copyright 2018 Ericsson AB.
+   For a full list of individual contributors, please see the commit history.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+--->
+
+# Contributing to Eiffel Community
+Thank you for contributing to the Eiffel Community!
+
+The following is a set of guidelines for contributing to this repository.
+
+## How to Propose Changes
+Anyone is welcome to propose changes to the contents of this repository by creating a new [Issue](https://github.com/eiffel-community/eiffel/issues) ticket in GitHub. These requests may concern anything contained in the repo: changes to documentation, bug fixes, requests for additional information, additional event types, requests for additional examples, new featuers et cetera.
+
+When posting a new issue, try to be as precise as possible and phrase your arguments for your request carefully. Keep in mind that collaborating on software development is often an exercise in finding workable compromises between multiple and often conflicting needs; this is particularly true for defining a shared protocol such as Eiffel. In particular, pay attention to the following:
+1. What type of change is requested?
+1. Why would you like to see this change?
+1. Can you provide any concrete examples?
+1. Which arguments in favor can you think of?
+1. Which arguments against can you think of, and why are they outweighed by the arguments in favor?
+
+Also, keep in mind that just as anyone is welcome to propose a change, anyone is welcome to disagree with and criticize that proposal.
+
+### Closing Issues
+An Issue can be closed by any member of [the repository maintainers' team](https://github.com/orgs/eiffel-community/teams/eiffel-protocol-maintainers). This can happen in various ways, for varying reasons:
```

Comment:
This link is broken, for me at least.

Reply:
This is something we need to make it more visible - who are the maintainers and how to communicate with them so people can easily find that info.

I'll create a new issue and propose using Github code owners file/feature to increase the visibility of the maintainers and also notify those maintainers for the new PRs.

https://help.github.com/en/articles/about-code-owners

Reply:
Yes, using the code owners file make sense (although I'm not sure I understand all possibilities in using it).

Reply:
Some open source projects keep a file in their repositories and list maintainers, meeting logistics, and other project specific information.

Codeowners file seems to be the closest thing to that and helps us not to invent the mechanism ourselves. Other good thing about this file is that Github notifies those people for new PRs so people don't need to send mails etc for getting attention from the maintainers.

"**Code owners are automatically requested for review when someone opens a pull request that modifies code that they own.**"